app/scripts/app.jsx

```diff
@@ -47,14 +48,17 @@ let CreateSchedule = React.createClass({
 
   handleSubmit(e) {
     e.preventDefault();
+    let id = Date.now().toString();
     let map = Array.prototype.map;
     let uniqueDashboardUrl = map.call(Date.now().toString(), function(digit) { return (10 - digit).toString(); }).join('');
 
     let schedule = {
-      url: Date.now().toString(),
+      url: id,
       dashboardUrl: uniqueDashboardUrl,
       slots: this.props.timeSlots,
     }
+
+    Schedules[id] = schedule;
   },
```

Comment:
What's the purpose of this line of code? If we want to save the state of a schedule, then we have to do it on the server-side. This on the client-side, so it's unique to the browser and will be lost if they close the tab or navigate to another page.

You need to get this data over the great wall to the server to persist. That can be done any one of the following ways:

- Send an HTTP post request with the information
- Send an AJAX request
- Send the information over WebSockets

Options one and two are probably the best in this case because it will be less work to handle any errors, but it—frankly—wouldn't be too bad with option three either.